crates/rover-client/src/introspection/schema.rs

```diff
@@ -0,0 +1,237 @@
+//! Schema encoding module used to work with Introspection result.
+//!
+//! More information on Schema Definition language(SDL) can be found in [this
+//! documentation](https://www.apollographql.com/docs/apollo-server/schema/schema/).
+//!
+use crate::query::graph::introspect;
+use sdl_encoder::{
+    Directive, EnumDef, EnumValue, Field, FieldValue, InputObjectDef, InputValue, InterfaceDef,
+    ObjectDef, ScalarDef, Schema as SDL, UnionDef,
+};
+use serde::Deserialize;
+use std::convert::TryFrom;
+
+pub type Introspection = introspect::introspection_query::ResponseData;
+pub type SchemaTypes = introspect::introspection_query::IntrospectionQuerySchemaTypes;
+pub type SchemaDirectives = introspect::introspection_query::IntrospectionQuerySchemaDirectives;
+pub type FullTypeFields = introspect::introspection_query::FullTypeFields;
```

Comment:
Why are these types plural if they represent a single object? Wouldn't `SchemaType`, `SchemaDirective`, and `FullTypeField` be more appropriate? Or perhaps `IntrospectionType`, `IntrospectionDirective`, and `IntrospectionField` (those are some of [the types used in `graphql-js` to represent the result of an introspection query](https://github.com/graphql/graphql-js/blob/main/src/utilities/getIntrospectionQuery.js#L153-L154))?

Reply:
ah yea, thanks, these were just autocompleted from the generated `introspection-query` struct, and the names there are in plural.